### PR TITLE
$insertNodes: Selection-agnostic node insertion with Grid/Node selection support

### PR DESCRIPTION
**Pull request by zurfyx**
Node selection

https://user-images.githubusercontent.com/193447/178761488-d39987f7-2a90-4604-b4fc-dfbf57178f35.mov

Grid selection

https://user-images.githubusercontent.com/193447/178761618-316b267b-2105-48d5-9714-4a9268e0fc1b.mov

### CODE REVIEW

File: packages/lexical-playground/src/plugins/ToolbarPlugin.tsx [hunk 1]
```diff
@@ -426,6 +425,19 @@ function InsertImageDialog({
   onClose: () => void;
 }): JSX.Element {
   const [mode, setMode] = useState<null | 'url' | 'file'>(null);
+  const hasModifier = useRef(false);
+
+  useEffect(() => {
+    hasModifier.current = false;
+    const handler = (e: KeyboardEvent) => {
+      // debugger;
```
Comment (acywatson) [inline packages/lexical-playground/src/plugins/ToolbarPlugin.tsx:433]: whoops

File: packages/lexical/src/LexicalNormalization.ts [hunk 2]
```diff
@@ -84,3 +84,65 @@ export function $normalizeTextNode(textNode: TextNode): void {
     }
   }
 }
+
+export function $normalizeSelection(selection: RangeSelection): RangeSelection {
```
Comment (acywatson) [inline packages/lexical/src/LexicalNormalization.ts:88]: Do we need a whole separate module in core for this? What else will go in here?

**Comment by acywatson** [comment]
This looks awesome! Planning to step through this logic a bit more later, maybe a comment or two on the normalization logic would be good?

**Review by thegreatercurve — APPROVED** [review]

File: packages/lexical/src/LexicalNormalization.ts (hunk 2 above)
Comment (acywatson) [reply packages/lexical/src/LexicalNormalization.ts:89]: Oh lol I never knew we had a LexicalNormalization module so I assumed you created it. #til

**Review by acywatson — APPROVED** [review]

**Comment by acywatson** [comment]
@zurfyx ?

**Comment by acywatson** [comment]
> @zurfyx ?

Also stop using that podcast tweet as test data lol

**Pull request merged**